### PR TITLE
Pipelines Configuration: Parameter options overrides (DFCT0010053)

### LINKED ISSUE #592: Pipelines Configuration: Parameter options overrides
## Problem

As an administrator of IRIDA Next I would like to be able to provide my own sets of options for pipeline parameters when I register a pipeline in the system. Specifically I would like to be able to provide a set of database options which would render a select dropdown for a specified database option with a name (visible to the user) and a path which is passed to the pipeline. This would ensure that my Users would be able to launch a pipeline with a given database without having knowledge of our database store. These parameter overrides could be set for a specific version of a pipeline or for any version of the same pipeline.

## Summary

* Update pipeline json config at `config/pipelines/pipelines.json` to support overriding parameters in `version` config or in pipeline entry.
  ```json
  [
    {
      "url": "https://github.com/phac-nml/iridanextexample",
      "name": "phac-nml/iridanextexample",
      "description": "IRIDA Next Example Pipeline",
      "parameter_overrides": {
         "database": {
           "description": { # can be left for later
             "en": "Englush fdescpt",
             "fr": "Francais safd"
           },
           "default": "DBNAME",
           "enum": [
            {"DBNAME": "PATH_TO_DB"}
           ]
         }
      },
      "versions": [
        {
          "name": "1.0.2",
          "parameter_overrides": {
            "database": {
              "default": "DBNAME",
              "enum": [
                {"DBNAME": "PATH_TO_DB"}
              ]
            }
          },
        },
      ],
    },
  ]
  ```
* Update `lib/irida/pipelines.rb` to parse the section for the pipeline in `register_pipelines` to parse `parameter_overrides` and merge global if present with version specific `parameter_overrides`
* Update `lib/irida/pipeline.rb` to include `parameter_overrides` attr and method to access it
* Update `NextflowComponent` to check `parameter_overrides` and if present modify parameter for display

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Fixes #592 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

In  `config/pipelines/pipelines.json` add mikrokonda:

```json
{
    "url": "https://github.com/phac-nml/mikrokondo",
    "name": "phac-nml/mikrokondo",
    "description": "Mikrokondo pipeline",
    "parameter_overrides": {
      "definitions": {
        "databases_and_pre_computed_files": {
          "properties": {
            "kraken2_db": {
              "type": "string",
              "description": "Kraken2 database",
              "enum": [
                [
                  "DBNAME",
                  "PATH_TO_DB"
                ],
                [
                  "ANOTHER_DB",
                  "ANOTHER_PATH"
                ]
              ]
            }
          }
        }
      }
    },
    "versions": [
      {
        "name": "0.1.2"
      }
    ]
  }
```

After selecting the pipeline, it should render:

image

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/components/nextflow_component.rb [hunk 1]
```diff
@@ -18,4 +18,14 @@ def initialize(url:, samples:, workflow:, fields:, namespace_id:, allowed_to_upd
   end
 
   # rubocop:enable Metrics/ParameterLists
+
+  def text_for_description(description)
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/components/nextflow_component.rb:22]: Can we call this method or rename it to work for pipeline description and section title and description, e.g.:

image

**Review by ericenns — CHANGES_REQUESTED** [review]
This is looking good, just a few comments.

File: app/components/nextflow_component.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshsadam) [reply app/components/nextflow_component.rb]: Updated in 1093612404b966c7caf10df452d43ad0449d3495

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**